plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerSplitSource.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.faker;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.spi.HostAddress;
+import io.trino.spi.connector.ConnectorSplit;
+import io.trino.spi.connector.ConnectorSplitSource;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.LongStream;
+
+import static java.lang.Math.ceilDiv;
+import static java.lang.Math.min;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+
+public class FakerSplitSource
+        implements ConnectorSplitSource
+{
+    // this is equal to 10k pages generated in FakerPageSource
+    static final long MAX_ROWS_PER_SPLIT = 10_000L * 4096;
+
+    private final long splitLimit;
+    private final long lastSplitLimit;
+    private final Iterator<HostAddress> splits;
+
+    public FakerSplitSource(List<HostAddress> addresses, long minSplitsPerNode, long tableLimit)
+    {
+        requireNonNull(addresses, "addresses is null");
+        long splitsPerNode = addresses.size() * minSplitsPerNode;
+        this.splitLimit = min(ceilDiv(tableLimit, splitsPerNode), MAX_ROWS_PER_SPLIT);
+        splitsPerNode = ceilDiv(tableLimit, splitLimit * addresses.size());
+        this.lastSplitLimit = tableLimit - splitLimit * (addresses.size() * splitsPerNode - 1);
+        this.splits = LongStream.range(0, splitsPerNode)
+                .boxed()
+                .flatMap(_ -> addresses.stream())
+                .iterator();
```

Comment:
```
int splitsCount = IntMath.divide(tableLimit, MAX_ROWS_PER_SPLIT, RoundingMode.UP);
ImmutableList.Builder<ConnectorSplit> splits = ImmutableList.builder();
for (int i = 0; i < splitCount - 1; i++) {
     HostAddress node = addresses.get(i % addresses.size());
     splits.add(new FakerSplit(node, MAX_ROWS_PER_SPLIT);
}
HostAddress node = addresses.get((splitCount - 1) % addresses.size());
splits.add(new FakerSplit(node, tableLimit % MAX_ROWS_PER_SPLIT);
return new FixedSplitSource(splits.build());
```
We don't need FakerSplitSource, can do this directly in getSplits

Reply:
I did this to avoid creating too large lists of splits at once. But I admit I don't know the practical limit. Is there any, if generating this list is quick enough?

Reply:
After adjusting `MAX_ROWS_PER_SPLIT` to 250 * 4096, when generating 100_000_000 rows on a cluster with 100 nodes, we'd have about 10k splits. I'll move it back to the split manager, thanks!

Reply:
All done, thanks! I kept it in a separate commit, but that's only to make review easier. LMK if I should squash it.